eng/pipelines/linux.yml

```diff
@@ -39,6 +39,15 @@ jobs:
             _buildScriptPrefix: 'ROOTFS_DIR=/crossrootfs/arm64 '
             _buildExtraArguments: -warnAsError false
 
+          musl_x64_Release:
```

Comment:
Instead of making these to run in both, CI and official builds, let's cover both `Debug` and `Release` flavors. 

CI -> `Debug` with tests.
Official -> `Release` with tests.

Delete `musl_x64_Debug` from `LinuxNoTest` job: https://github.com/dotnet/corefx/pull/36392/files#diff-6be2d586a9ac2642857a1e750a752e02R113